### PR TITLE
Feature/lavaut m ccheat

### PR DESCRIPTION
**Pull request by emilelavaut**
Last version of the singleHit module to identify isolated blips as radiologicals for calibration.

I've add a generator tag to hit matching in order to evaluate purity and efficiency of my analysis

### CODE REVIEW

File: protoduneana/verticaldrift/singlehit/SingleHit_module.cc
Comment (tomjunk) [inline protoduneana/verticaldrift/singlehit/SingleHit_module.cc]: There are a lot of std::cout statements in here that look like they might be useful for debugging but will just clutter up logfiles. Is this module meant to be run in production? The message facility allows fcl steering of log levels and throttles many repeated messages, but sometimes I too prefer std::cout, if only to know that messages haven't been throttled. But ... Messages like these do not give the reader of a logfile any clue what piece of code wrote the message. If someone wants to understand what a particular message means, they have to look through all of DUNE code and LArSoft for the character string in a message. So please write the method name as part of the message, and make them optional.

File: protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl
Comment (tomjunk) [inline protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl:34]: services.Geometry.GDML, not GDLM

File: protoduneana/verticaldrift/singlehit/SingleHit_module.cc
Comment (tomjunk) [reply protoduneana/verticaldrift/singlehit/SingleHit_module.cc]: And there's a grammar error "IS ... POINTS"  (number mismatch).

File: protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl
Comment (tomjunk) [inline protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl:55]: And a suggestion on fcl configurations -- since you have three run*.fcl files with similar parameter choices, you should make a shared fcl file with 

BEGIN_PROLOG 
  SingleHItCheaterDefaults: {
  }
END_PROLOG

and then #include that defaults file and have the line

ana: @local::SingleHItCheaterDefaults

Then you can override any default parameters in the individual run files.  This helps centralize maintenance in case you add a parameter someday, and it helps people run your module in their jobs and not just in your run* fcl jobs.

Comment (tomjunk) [reply protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl:55]: See for example the Kalman filter track fitter default parameter fcl file for the pattern
https://github.com/LArSoft/larreco/blob/develop/larreco/TrackFinder/kalmanfilterfinaltrackfitter.fcl

File: protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc
Comment (aolivier23) [inline protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc:259]: This might set off Clang when we try to cut a release.  `tfs` looks like an unused variable to me.  Can you remove it?

File: protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc
Comment (aolivier23) [inline protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc:109]: This might set off Clang when we try to cut a release.  `detProp` looks like an unused variable to me.  Can you remove it without changing what the code does?

**Comment by emilelavaut** [comment]
Thank you for your comments,
I'have put so Loglevel condition before every std::cout and changed the fcl

For the  comment on the MCCheater I also deleted the non-used objects but the module is just an intermediate module to understand the cheater not to use at production levels

**Comment by tomjunk** [comment]
Okay, thanks for the commits and bug fixes.  I think you still have some GDLMs in there:

in protoduneana with your PR checked out:

grep -r -i gdlm *
verticaldrift/singlehit/runMCcheat_PDHD.fcl:services.Geometry.GDLM: "protodunehd_v6_refactored.gdml"
verticaldrift/singlehit/runMCcheat_PDVD.fcl:services.Geometry.GDLM: "protodunevd_v4_refactored.gdml"

**Comment by emilelavaut** [comment]
Thank you, I fixe that now !

**Comment by sam-fogarty** [comment]
Is it possible to merge this PR soon? @tomjunk @aolivier23 ?

**Comment by aolivier23** [comment]
I'll try to get it into a new release today.  It seemed just about ready the last time I looked at it.  Taking a look at the changes before I commit to that.

**Review by aolivier23 — COMMENTED** [review]
Looks good to me.  I'm of the opinion that we should move on with doing great physics.  I'll approve as soon as I make sure it passes tests.

In the future, we could improve:
- Consider changing std::cout and LogLevel to MF_LOG_DEBUG(), mf::LogInfo(), etc. cout can be good for avoiding the automatic message squasher, but it's not clear to me that the message squasher would get in your way.

**Comment by aolivier23** [comment]
trigger build with DUNE/protoduneana#34

**Comment by aolivier23** [comment]
Looks like I can fix the things that caused the build to fail.  Just an integer sign mismatch and a typo in a variable name.  Bear with me while I learn how to do this.  I don't think it needs any action from you.

**Comment by emilelavaut** [comment]
thank you !  yes I made some modification on github today and I was planning to test to build it to see the errors on monday. I was not expecting you to merge today sorry.

**Comment by aolivier23** [comment]
@emilelavaut No problem.  Do you want me to wait until the next release to merge this pull request?  I'm going to cut a release today to match the new LArSoft version.  Sounds like the next release will be next Friday.

**Comment by sam-fogarty** [comment]
@aolivier23 @emilelavaut I may suggest waiting until next Friday as there is at least one change we are going to add (getting absolute timestamps for the clusters) which is relevant for charge -light matching. So we might as well wait, and maybe Emile will have a few more small changes he may want to add.

**Comment by emilelavaut** [comment]
Yes next friday would be perfect for me !

**Comment by emilelavaut** [comment]
this version of the code work well and have everything updated.
it has been test and it should be able to merge.

**Comment by aolivier23** [comment]
Thanks @emilelavaut .  I'm going to try to merge this PR in tomorrow afternoon.  Speak now if you want me to hold off please!

**Comment by aolivier23** [comment]
trigger build with dune/protoduneana#34

**Review by aolivier23 — APPROVED** [review]
Looks good!  A few FHICL files changed to resolve Tom's comment about using an include file.  Very readable now.

Note that I resolved a merge conflict in the module.  I updated a function name to match the new LArSoft version's breaking change.

**Pull request merged**
